api_draft.txt

```diff
@@ -0,0 +1,128 @@
+
+################################################################################
+Arimaa Game Server JSON API
+################################################################################
+
+#TODO - see todos below
+
+All requests and responses are in JSON.
+
+Shared types:
+<Message> = {"message":<String>}
+<Error> = {"error":<String>}
+<GameID> = <Int>   #Identifier for a game
+<Auth> = <String>  #Authorization token returned from logging in
+<GameAuth> = <String>  #Authorization token returned from joining a game
+<Username> = <String>
+<Timestamp> = <Float> #Seconds since the epoch (1970-01-01 00:00 UTC) (up to however much precision the server has)
+
+===ACCOUNTS========================================================
+To register an account:
+POST /accounts/register {"username":<Username>,"email":<String>,"password":<String>}
+
+#TODO add other account facilities
+#TODO add ways of looking up users
+
+===LOGIN===========================================================
+To log in:
+POST /accounts/login {"username":<Username>,"password":<String>}
+Response if successful = {"username":<Username>,"auth":<Auth>}
+Response if not successful = <Error>
+
+To log out:
+POST /accounts/logout {"auth":<Auth>}
+Response if successful = <Message>
+Response if not successful = <Error>
+
+===GAMEROOM========================================================
+<Board> = {
+  "history":[<String>,<String>,<String>,...], #Move history in the order [1g,1s,2g,2s...] ex: ["Ra1","Ra7","Ra1n Ra2n Ra3n Ra4n"]
+  "toMove":<String>,  #Next player to move, either "g" or "s"
+  "position":<String> #64-char string specifying the board in the order a8-h8,a7-h7,...,a1-h1
+                      #ex:"rrrrrrrrchdemdhc                                CHDMEDHCRRRRRRRR"
+}
+
+<UserInfo> = {
+  "username":<Username>,
+  "rating":<Float>
+}
+
+<GameResult> = {
+  "resultCode":<String>, #TODO - describe this. Same as current Arimaa resultCode system seems okay
+  "endTime":<Timestamp>
+}
+
+<PlayerTimeControl> = {
+  #TODO - probably just adopt the current arimaa time control system or something similar
+}
+
+<GameMetadata> = {
+  "id":<Int>,
+  "numMoves":<Int>, #Total number of moves made in the game (ex: {1g, 1s, 2g, 2s, 3g} = 5 moves)
```

Comment:
I would call these 'halfmoves'. I think most people would think a move after both players play. This is also what its called in chess.